### PR TITLE
Telemetry operator

### PR DESCRIPTION
**Pull request by paramite**
Integration of OpenStack Telemetry services. Adds CeilometerCentral and CeilometerCompute resources via Telemetry resource and manager.

### CODE REVIEW

**Comment by abays** [comment]
CI is currently failing because the PR needs to `go get` the latest Telemetry operator bits (https://github.com/openstack-k8s-operators/telemetry-operator/pull/88).

File: main.go [hunk 1]
```diff
@@ -304,4 +306,16 @@ func setupServiceOperatorDefaults() {
...
 	redisv1.SetupRedisDefaults(redisDefaults)
+
+	// Telemetry
+	telemetryDefaults := telemetryv1.TelemetryDefaults{
+		CentralContainerImageURL:      os.Getenv("CEILOMETER_CENTRAL_IMAGE_URL_DEFAULT"),
+		CentralInitContainerImageURL:  os.Getenv("CEILOMETER_CENTRAL_INIT_IMAGE_URL_DEFAULT"),
+		ComputeContainerImageURL:      os.Getenv("CEILOMETER_COMPUTE_IMAGE_URL_DEFAULT"),
+		ComputeInitContainerImageURL:  os.Getenv("CEILOMETER_COMPUTE_INIT_IMAGE_URL_DEFAULT"),
+		NotificationContainerImageURL: os.Getenv("CEILOMETER_NOTIFICATION_IMAGE_URL_DEFAULT"),
+		SgCoreContainerImageURL:       os.Getenv("CEILOMETER_SGCORE_INIT_IMAGE_URL_DEFAULT"),
```
Comment (abays) [inline main.go:317]: Bad copy-paste here of a mistake I made in the initial Telemetry operator webhook PR.  Need to fix this: `s/CEILOMETER_SGCORE_INIT_IMAGE_URL_DEFAULT/CEILOMETER_SGCORE_IMAGE_URL_DEFAULT`

**Comment by jlarriba** [comment]
This was an error accessing the registry:

error: build error: error creating build container: initializing source docker://quay.io/openstack-k8s-operators/ironic-operator-bundle:a6ec1284c18bda43dad32b36afac446a36dd5603: too many requests to registry

/test openstack-operator-build-deploy

**Comment by jlarriba** [comment]
I have not being paying the required atention to the statuses lately, I will fix this issue with CeilometerCompute ASAP.

**Comment by jlarriba** [comment]
@abays that would be utterly useful, yes

**Comment by abays** [comment]
Now that https://github.com/openstack-k8s-operators/telemetry-operator/pull/92 has merged, we should incorporate it into this patch to see if it helps us pass CI.

```
go get -u github.com/openstack-k8s-operators/telemetry-operator/api
cd apis
go get -u github.com/openstack-k8s-operators/telemetry-operator/api
cd ..
make tidy
```

**Comment by abays** [comment]
> @paramite: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> Test name 	Commit 	Details 	Required 	Rerun command
> ci/prow/openstack-operator-build-deploy 	[0ca4f5b](https://github.com/openstack-k8s-operators/openstack-operator/commit/0ca4f5b60cb819bffd5a3c2c767961e705c69e43) 	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/309/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1659325878884110336) 	true 	`/test openstack-operator-build-deploy`
> 
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openstack-k8s-operators&repo=openstack-operator&pr=309). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:paramite).

It looks like it failed to build the OpenStack operator.  Did not even get to the deployment testing.  Maybe a hiccup?  Will try again.

/test openstack-operator-build-deploy

**Comment by abays** [comment]
> > @paramite: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> > Test name 	Commit 	Details 	Required 	Rerun command
> > ci/prow/openstack-operator-build-deploy 	[0ca4f5b](https://github.com/openstack-k8s-operators/openstack-operator/commit/0ca4f5b60cb819bffd5a3c2c767961e705c69e43) 	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/309/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1659325878884110336) 	true 	`/test openstack-operator-build-deploy`
> > [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openstack-k8s-operators&repo=openstack-operator&pr=309). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:paramite).
> 
> It looks like it failed to build the OpenStack operator. Did not even get to the deployment testing. Maybe a hiccup? Will try again.
> 
> /test openstack-operator-build-deploy

I have reproduced the problem locally.  Investigating now.

**Comment by abays** [comment]
> > > @paramite: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> > > Test name 	Commit 	Details 	Required 	Rerun command
> > > ci/prow/openstack-operator-build-deploy 	[0ca4f5b](https://github.com/openstack-k8s-operators/openstack-operator/commit/0ca4f5b60cb819bffd5a3c2c767961e705c69e43) 	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/309/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1659325878884110336) 	true 	`/test openstack-operator-build-deploy`
> > > [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openstack-k8s-operators&repo=openstack-operator&pr=309). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:paramite).
> > 
> > 
> > It looks like it failed to build the OpenStack operator. Did not even get to the deployment testing. Maybe a hiccup? Will try again.
> > /test openstack-operator-build-deploy
> 
> I have reproduced the problem locally. Investigating now.

The build GitHub Action for the Telemetry operator had failed due to a hiccup.  The images are available now, so let's re-run the job here.

/test openstack-operator-build-deploy

**Comment by abays** [comment]
@paramite @jlarriba Deploy job is green \o/

**Review by abays — COMMENTED** [review]
+1 from me.  Just need to hear from @dprince if this should land in the main OpenStack operator bundle, which it currently will as-is.  I checked the CRD and CSV size increase to include it and they look like so:

```
$ ls -lh csv*
-rw-rw-r--. 1 ocp ocp 301K May 19 12:24 csv_normal.out
-rw-rw-r--. 1 ocp ocp 319K May 19 12:30 csv_telemetry.out

$ ls -lh crd*
-rw-rw-r--. 1 ocp ocp 609K May 19 12:26 crd_normal.out
-rw-rw-r--. 1 ocp ocp 614K May 19 12:30 crd_telemetry.out
```

**Comment by abays** [comment]
@paramite It looks like this needs a `make tidy`:

```
Error: err: exit status 1: stderr: go: updates to go.mod needed; to update it:
	go mod tidy
```

**Comment by paramite** [comment]
@abays @dprince  Anything else?

**Comment by abays** [comment]
@paramite I think we can merge this once it's rebased and passes CI.  If we need to split it out into a separate bundle, we can do that later.  You may be wondering what that means, so in summary: there is a concern with overloading this operator's CSV and CRD such that they become too large and cannot be processed.  Whenever we integrate a new operator into this operator, the CSV and the CRD sizes increase.  As I noted in a previous comment though [1], it seems okay for now.

[1] https://github.com/openstack-k8s-operators/openstack-operator/pull/309#pullrequestreview-1434716646

**Review by abays — APPROVED** [review]
/lgtm

**Comment by mrunge** [comment]
thank you very much here!

**Pull request merged**